docs/porting/libraries.md

```diff
@@ -1,5 +1,224 @@
+
+This approach is a bit more careful and structured than simply working out compiler errors, but it is still relatively code-focused and has the benefit of always having code that can compile.  The way you resolve certain issues that couldn't be addressed by just using another API can vary greatly.  You may find that you need to develop a more comprehensive plan for certain projects, which is covered as the next approach.
+
+### Developing a comprehensive plan of attack
+
+This approach may be best for larger and more complex projects, where restructuring of code or rewriting certain areas may be necessary to support .NET Core.  The approach is as follows:
+
+1. Run API Port on a project.
+2. Understand where in your code each non-portable type is being used and how that affects overall portability.
+
+   a.  Understand the nature of those types.  Are they small in number, but used frequently?  Are they large in number, but used infrequently?  Is their use concentrated, or is it spread throughout your code?
+   
+   b. Is it easy to isolate code that isn't portable so you can deal with it easier?
+   
+   c. Would you need to refactor your code?
+   
+   d. For these types which aren't portable, are there alternative APIs that accomplish the same task?  For example, if you're using `WebClient`, you may be able to use `HttpClient` instead.
+   
+   e. Are there different portable APIs you can use to accomplish a task, even if it's not a drop-in replacement?  For example, if you're using `XmlSchema` to help parse XML but you don't require XML schema discovery, you could use Linq to XML APIs and hand-parse the data.
+
+3. If you have assemblies that are difficult to port, is it worth leaving them on .NET Framework for now?
+
+   a. You may have some functionality in your library that's incompatible with .NET Core because it relies too heavily on .NET Framework- or Windows-specific functionality.  Is it worth leaving that functionality behind for now and releasing a .NET Core version of your library with less features for the time being?
+   
+   b. Would a refactor help here?
+   
+4. Is it reasonable to write some code a .NET Framework API was doing yourself?
+
+   a. You could consider copying, modifying, and using code from the [.NET Framework Reference Source](https://github.com/Microsoft/referencesource).  It's licensed under the [MIT License](https://github.com/Microsoft/referencesource/blob/master/LICENSE.txt) so you have significant freedom in doing this.  Just make sure to properly attribute Microsoft in your code!
+   
+5. Repeat this process as needed for different projects.
+6. Once your have a plan, execute that plan.
+ 
+The analysis phase could take some time depending on how large your codebase is.  But that's okay!  Spending time in this phase to thoroughly understand the scope of changes needed and to develop a plan can save you a lot of time in the long run, particularly if you have a more complex codebase.
```

Comment:
I'd remove "But that's okay!" because if a person is stressed and under time pressure, it might not be okay for them.